test/e2e/serial/tests/resource_hostlevel.go

```diff
@@ -321,14 +321,19 @@ var _ = Describe("[serial][hostlevel] numaresources host-level resources", Seria
 				err := fxt.Client.Create(ctx, pod)
 				Expect(err).NotTo(HaveOccurred(), "unable to create test pod %q", pod.Name)
 
-				var updatedPod corev1.Pod
-				err = fxt.Client.Get(context.TODO(), client.ObjectKey{Namespace: pod.Namespace, Name: pod.Name}, &updatedPod)
-				Expect(err).ToNot(HaveOccurred())
+				updatedPod, err := wait.With(fxt.Client).Timeout(time.Minute).ForPodPhase(ctx, pod.Namespace, pod.Name, corev1.PodPending)
+				if err != nil {
+					_ = objects.LogEventsForPod(fxt.K8sClient, updatedPod.Namespace, updatedPod.Name)
+				}
+				Expect(err).NotTo(HaveOccurred(), "Pod %s/%s was found  in state %q while expected to be Pending", updatedPod.Namespace, updatedPod.Name, updatedPod.Status.Phase)
```

Comment:
Or move the `get` into a `Eventually`. I like more the current approach you take, so let's go with it.